### DISCUSSION TITLE
question about data type of parameter ob

### DISCUSSION THREAD

**Discussion by ww95211** [resiprocate/resiprocate#198] [Ideas]
I notice the data type of parameter ob is defined as ExistsParameter, but in RFC5626,  the parameter ob is used to indicate a flow token in section 2.1, that means  the parameter should has its value if it exists in Record-Route. So my question is , should be the data type of parameter ob should defined as ExistsOrDataParameter?

**Comment by sgodin** [comment]
The ;ob parameter is an "exists" parameter only.  It is used to indicate
that the Path or Route header URI contains a flow token.  It doesn't
specify the actual flow token itself.  See the sample messages on the pages
in the 30's of the RFC to see how it is used.

Best Regards,
Scott Godin

**Comment by ww95211** [comment]
thanks @sgodin
